### PR TITLE
Greetings flag

### LINKED ISSUE #3: [Flag] add greetings flag
Greetings flag requires a Component FlagValue 

- when a user enters a zone with the greetings flag enabled, then send the attached message
- add the command ``zone flags <zone> greetings message set [message]``
- add the command ``zone flags <zone> greetings view``
- add the command ``zone flags <zone> greetings remove message``
- permission to create own ``zone.region.flag.greetings.own.create``
- permission to remove own ``zone.region.flag.greetings.own.remove``
- permission to create others
- permission to remove others
By not specifying a message, it wont send a message to the user (not removing the flag)

### PR DESCRIPTION
**Pull request by mosemister**

### CODE REVIEW

**Comment by Lukamon24** [comment]
yoyoyo wot

File: src/main/java/org/zone/commands/structure/ZoneCommands.java [hunk 1]
```diff
@@ -47,7 +48,7 @@
     ZoneFlagBlockPlaceSetGroupCommand ZONE_FLAG_BLOCK_PLACE_SET_GROUP_COMMAND = new ZoneFlagBlockPlaceSetGroupCommand();
     ZoneViewBalanceCommand ZONE_FLAG_VIEW_BALANCE_COMMAND = new ZoneViewBalanceCommand();
     SetGreetingsMessageCommand SET_GREETINGS_MESSAGE_COMMAND = new SetGreetingsMessageCommand();
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/ZoneCommands.java]: Could the class be renamed to ``ZoneFlagGreetingsSetMessageCommand`` to make it consistant with the others

File: src/main/java/org/zone/commands/structure/ZoneCommands.java [hunk 2]
```diff
@@ -47,7 +48,7 @@
...
-
+    GreetingsMessageViewCommand GREETINGS_MESSAGE_VIEW_COMMAND = new GreetingsMessageViewCommand();
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/ZoneCommands.java]: Could the class be renamed to ``ZoneFlagGreetingsViewCommand`` to make it consistant with the others

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import org.zone.region.flag.FlagTypes;
+import org.zone.region.flag.greetings.GreetingsFlag;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class GreetingsMessageViewCommand implements ArgumentCommand {
+    public static final ExactArgument ZONE = new ExactArgument("zone");
+    public static final ExactArgument FLAGS = new ExactArgument("flags");
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java]: Could you make the id ``flag`` to make it consistant.

Same in the set command too

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    public static final ZoneArgument ZONE_VALUE = new ZoneArgument("zone_value", new ZoneArgument.ZoneArgumentPropertiesBuilder());
+    public static final ExactArgument GREETINGS = new ExactArgument("greetings");
+    public static final ExactArgument MESSAGE = new ExactArgument("message");
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java]: You are not using MESSAGE. please remove this line

File: src/main/java/org/zone/commands/structure/ZoneCommands.java (hunk 1 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/ZoneCommands.java]: ok?

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java (hunk 3 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java]: wdym

File: src/main/java/org/zone/commands/structure/ZoneCommands.java (hunk 1 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/ZoneCommands.java]: DONE will update

File: src/main/java/org/zone/commands/structure/ZoneCommands.java (hunk 2 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/ZoneCommands.java]: DONE will update

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java (hunk 4 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java]: Done will update

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java (hunk 3 above)
Comment (mosemister) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java]: You see where it says ``"flags"`` could this be ``"flag"``?

Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/GreetingsMessageViewCommand.java]: done will update

**Pull request merged**